src/Microsoft.Health.SqlServer/DefaultSqlConnectionBuilder.cs

```diff
@@ -96,4 +99,13 @@ protected virtual SqlConnectionStringBuilder GetConnectionStringBuilder(string i
 
         return builder;
     }
+
+    /// <summary>
+    /// Creates a <see cref="SqlConnectionStringBuilder"/> using the configured connection string and modified based on the input.
+    /// </summary>
+    /// <param name="initialCatalog">An optional initial catalog that may be used to override the default value.</param>
+    /// <param name="maxPoolSize">An optional maximum for the SQL connection pool size.</param>
+    /// <returns>A <see cref="SqlConnectionStringBuilder"/> representing the current connection string.</returns>
+    protected virtual async Task<SqlConnectionStringBuilder> GetConnectionStringBuilderAsync(string initialCatalog = null, int? maxPoolSize = null)
+        => await Task.FromResult(GetConnectionStringBuilder(initialCatalog, maxPoolSize)).ConfigureAwait(false);
```

Comment:
nit: can remove `async` and `await` keywords here and instead just return the `Task`